templates/common/_base/files/mtu-migration-dispatcher.yaml

```diff
@@ -0,0 +1,295 @@
+
+        # Retrieve routes MTU and take the minimum (if there's any route MTU, we're probably
+        # interrupting an ongoing MTU migration and beginning a new one)
+        echo "phase1 starts"
+
+        min_route_mtu=""
+        while IFS= read -r line; do
+            route_mtu=$(echo $line | sed -n -e 's/^.*mtu \([^ ]*\).*/\1/p')
+            echo "mtu=$route_mtu in this route: $line"
+            if [ -n "$route_mtu" ]; then
+                if [ -z "$min_route_mtu" ]; then
+                    min_route_mtu=$route_mtu
+                fi
+                if [ $route_mtu -lt $min_route_mtu ]; then
+                    min_route_mtu=$route_mtu
+                fi
+            fi
+        done <<< "$iface_routes"
+
+        echo "min_route_mtu=$min_route_mtu"
+
+        max_dev_mtu=$iface_mtu
+
+        # same as above for parent dev
+        if [ -n "$parent_dev" ]; then
+            min_parent_route_mtu=""
+
+            while IFS= read -r line; do
+                parent_route_mtu=$(echo $line | sed -n -e 's/^.*mtu \([^ ]*\).*/\1/p')
+                echo "mtu=$parent_route_mtu in this route: $line"
+                if [ -n "$parent_route_mtu" ]; then
+                    if [ -z "$min_parent_route_mtu" ]; then
+                        min_parent_route_mtu=$parent_route_mtu
+                    fi
+                    if [ $parent_route_mtu -lt $min_parent_route_mtu ]; then
+                        min_parent_route_mtu=$parent_route_mtu
+                    fi
+                fi
+            done <<< "$parent_routes"
+
+            echo "parent_min_route_mtu=$min_parent_route_mtu"
+
+            # use the lower MTU between iface routes and parent routes
+            if [ $parent_min_route_mtu -lt  $min_route_mtu ]; then
+                min_route_mtu=$parent_route_mtu
+            fi
+            # use the higher MTU between iface and its parent
+            if [ $parent_mtu -gt $max_dev_mtu ]; then
+                max_dev_mtu=$parent_mtu
+            fi
+        fi
+
+        # compute new MTU for iface and routes
+        all_mtus_sorted=$(echo "${TARGET_MTU} ${max_dev_mtu} ${min_route_mtu}" | tr " " "\n"| sort -g | grep "\S")
+        echo "all_mtus_sorted=$all_mtus_sorted"
+        min_mtu=$(echo "$all_mtus_sorted" | head -n1)
+        max_mtu=$(echo "$all_mtus_sorted" | tail -n1)
+        new_iface_mtu=$max_mtu
+        new_route_mtu=$min_mtu
+        echo "new_iface_mtu=$new_iface_mtu, new_route_mtu=$new_route_mtu"
+
+        # apply new MTU on iface(s) and routes
+        # first on the parent and its routes...
+        if [ -n "$parent_dev" ]; then
+            set_mtu_on_dev $iface $conn_type $new_iface_mtu
+            while IFS= read -r line; do
+                # add the new MTU to the route
+                # if the route already has an MTU, it'll override it
+                ip route change ${line} dev ${parent_dev} mtu ${new_route_mtu}
```

Comment:
I don't think we need to manage the parent routes at all. At least not for any traffic that is related to the cluster.